internal/ignition/ignition_test.go

```diff
@@ -607,3 +610,97 @@ var _ = Describe("Openshift cluster ID extraction", func() {
 		Expect(err.Error()).To(Equal("no ClusterID field in cvo-overrides file"))
 	})
 })
+
+var _ = Describe("Generator UploadToS3", func() {
+	var (
+		ctx          = context.Background()
+		ctrl         *gomock.Controller
+		mockS3Client *s3wrapper.MockAPI
+	)
+
+	generator := installerGenerator{
+		log:     log,
+		workDir: workDir,
+	}
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		mockS3Client = s3wrapper.NewMockAPI(ctrl)
+
+		generator.s3Client = mockS3Client
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	mockUploadFile := func() *gomock.Call {
+		return mockS3Client.EXPECT().UploadFile(gomock.Any(), gomock.Any(), gomock.Any())
+	}
+
+	mockUploadObjectTimestamp := func() *gomock.Call {
+		return mockS3Client.EXPECT().UpdateObjectTimestamp(gomock.Any(), gomock.Any())
+	}
+
+	It("disabled host", func() {
+		cluster.Hosts = []*models.Host{
+			{Status: swag.String(models.HostStatusDisabled)},
+		}
+		generator.cluster = cluster
+
+		mockUploadFile().Return(nil).Times(len(fileNames))
+		mockUploadObjectTimestamp().Return(true, nil).Times(len(fileNames))
+
+		Expect(generator.UploadToS3(ctx)).ShouldNot(HaveOccurred())
+	})
+
+	Context("cluster with known host", func() {
+		BeforeEach(func() {
+			hostID := strfmt.UUID(uuid.New().String())
+			cluster.Hosts = []*models.Host{
+				{ID: &hostID, Status: swag.String(models.HostStatusKnown), Role: models.HostRoleMaster},
```

Comment:
Can you add a second host, just to be thorough?

Reply:
Sure